src/Components/Shared/src/DefaultJSDataStream.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal sealed class DefaultJSDataStream : BaseJSDataStream
```

Comment:
Would it be possible to find a name other than *Default* for this? It would be ideal to have something more descriptive to indicate how it differs from the others.

Suggestion: `PullFromJSDataStream`

Would be good also to add a comment above this indicating more about it, e.g.:

```
/// <Summary>
/// A stream that pulls each chunk on demand using JavaScript interop. This implementation is used for
/// WebAssembly and WebView applications.
/// </Summary>
```